ipyparallel/controller/hub.py

```diff
@@ -862,14 +873,13 @@ def save_iopub_message(self, topics, msg):
             name = content['name']
             s = '' if rec is None else rec[name]
             d[name] = s + content['text']
-
         elif msg_type == 'error':
             d['error'] = content
         elif msg_type == 'execute_input':
             d['execute_input'] = content['code']
         elif msg_type in ('display_data', 'execute_result'):
             d[msg_type] = content
-        elif msg_type == 'status':
+        elif msg_type in ('status', 'shutdown_reply', ):
```

Comment:
Since you've added monitor_iopub_message above, we can move the check for message types that should be ignored up to that level, so save only actually saves the response.